src/modules/displayList.js

```diff
@@ -41,11 +76,101 @@ export default async function populateGrid() {
     comBtn.textContent = 'Comment';
     comBtn.classList.add('btn', 'comment-button');
     comBtn.classList.add('btn-info');
+    comBtn.setAttribute('name', pokemon.name);
+    comBtn.setAttribute('onclick', `window.testyFunct( '${pokeImg.src}' )`);
+    comBtn.setAttribute('id', pokemon.height);
 
     element.appendChild(imgContainer);
     element.appendChild(pokeName);
     element.appendChild(comBtn);
 
     pokemons.append(element);
+    element.append(pokeName);
+  });
+  const btn = document.querySelectorAll('button');
+
+  btn.forEach((ele) => {
+    ele.addEventListener('click', (e) => {
+      detailSection.innerHTML = '';
+      document.querySelector('.bg-popup').style.display = 'flex';
+      pokeId = e.target.parentNode.id;
+      const title = document.createElement('h6');
+      title.innerHTML = `Name: ${ele.name}`;
+      detailSection.append(title);
+
+      const height = document.createElement('h6');
+      height.innerHTML = `height: ${ele.id}`;
+      detailSection.append(height);
+      commentsDOM();
+    });
   });
 }
+
+window.testyFunct = (e) => {
+  imageSection.innerHTML = '';
+  const img = document.createElement('img');
+  img.src = e;
+  imageSection.append(img);
+};
+
+const close = document.querySelector('.close');
+
+close.addEventListener('click', () => {
+  document.querySelector('.bg-popup').style.display = 'none';
+});
+
+const username = document.querySelector('#username');
+const comments = document.querySelector('#comments');
+const submit = document.querySelector('#submit');
+
+async function addcomments(e) {
+  e.preventDefault();
+  const endpoint = `apps/${APP_ID}/comments`;
+  const response = await fetch(BASE_URL + endpoint, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: pokeId,
+      username: `${username.value}`,
+      comment: comments.value,
+    }),
+
+  });
+  createComment();
+  commentsDOM();
+  return response;
+}
+
+submit.addEventListener('click', (event) => {
+  event.preventDefault();
+  addcomments(event);
+});
+
+// async function createComment() {
+//   const endpoint = `apps/${APP_ID}/comments?item_id=${pokeId}`;
+//   const response = await fetch(BASE_URL + endpoint);
+//   const data = await response.json();
+//   arrayOfComments = data;
+// }
+
+// async function commentsDOM() {
+//   await createComment();
+
+//   commentSection.innerHTML = '';
+//   if (arrayOfComments.length === 0) {
+//     commentsCount.innerHTML = '';
+//     commentSection.innerHTML = '';
+//   } else {
+//     arrayOfComments.forEach((comment) => {
+//       const p = document.createElement('p');
+//       p.innerHTML = `${comment.username}: ${comment.comment}`;
+//       p.className = 'comments';
+//       commentSection.append(p);
+//     });
+//     const count = arrayOfComments.length;
+//     commentsCount.innerHTML = count;
+//   }
+// }
```

Comment:
[Required ] Kindly delete this commented code in order to improve the file.  You could use git history instead.